Sources/ATProtoKit/Utilities/ATFacetParser.swift

```diff
@@ -95,6 +95,60 @@ public class ATFacetParser {
 
         return spans
     }
+    
+    /// Replaces URL text in the original string with a truncated version used for display in Bluesky.
+    /// This shortens each URL to exclude the scheme and limit to 32 characters total.
+    /// The replacement uses byte ranges, so the function processes from the end of the string backward.
+    public static func truncateAndReplaceLinks(in text: String) -> (text: String, updatedLinkFacets: [(url: URL, start: Int, end: Int)]) {
```

Comment:
```suggestion
    ///
    /// - Parameter text: The post's text.
    /// - Returns: A tuple containing the updated plain text and an array of link facets.
    ///   Each link facet represents a detected URL within the text, along with its start and end positions.
    public static func truncateAndReplaceLinks(in text: String) -> (text: String, updatedLinkFacets: [(url: URL, start: Int, end: Int)]) {
```

I'm okay with most of the code here, but the DocC complains that the properties and returns type have not been documented. Here's my suggestion.